R/cal-apply.R

```diff
@@ -0,0 +1,98 @@
+#---------------------------------- Apply --------------------------------------
+
+#' Applies a calibration to a set of prediction probabilities
+#' @details It currently supports data.frames only. It extracts the `truth` and
+#' the estimate columns names, and levels, from the calibration object.
+#' @param .data An object that can process a calibration object.
+#' @param calibration The calibration object (`cal_object`).
+#' @param ... Optional arguments; currently unused.
+#' @examples
+#' w_calibration <- cal_estimate_logistic(segment_logistic, Class)
+#'
+#' cal_apply(segment_logistic, w_calibration)
+#' @export
+cal_apply <- function(.data, calibration, ...) {
+  UseMethod("cal_apply")
+}
+
+#' @export
+cal_apply.data.frame <- function(.data, calibration, ...) {
+  if (calibration$type == "binary") {
+    cal_add_adjust(calibration, .data)
+  } else {
+    stop_multiclass()
+  }
+}
+
+# ------------------------------- Adjust ---------------------------------------
+
+cal_add_adjust <- function(calibration, .data) {
+  UseMethod("cal_add_adjust")
+}
+
+cal_add_adjust.cal_estimate_logistic <- function(calibration, .data) {
+  cal_add_predict_impl(
+    calibration = calibration,
+    .data = .data
+  )
+}
+
+cal_add_adjust.cal_estimate_logistic_spline <- function(calibration, .data) {
```

Comment:
I would just make a function

```r
foo<- function(calibration, .data) {
   cal_add_predict_impl(
     calibration = calibration,
     .data = .data
   )
 }
```

and assign it to each method.

Reply:
I did it like this so that `cal_apply()` only calls a single function, and then have the calibration class drive which implementation to use